nbodykit/algorithms/paircount_tpcf/tpcf.py

```diff
@@ -117,11 +121,19 @@ def run(self):
             # use the first randoms for both
             if self.data2 is not None and self.randoms2 is None:
                 self.randoms2 = self.randoms1
-
+            
             # use the Landy-Szalay estimator
             result = LandySzalayEstimator(pair_counter, self.data1, self.data2,
-                                            self.randoms1, self.randoms2, logger=self.logger, **attrs)
+                                            self.randoms1, self.randoms2, R1R2=self.R1R2,
+                                            logger=self.logger, **attrs)
             self.D1D2, self.D1R2, self.D2R1, self.R1R2, self.corr = result
+            
+            self.attrs['weightedpairs'] = {}
+            for pc in ['D1D2', 'D1R2', 'D2R1', 'R1R2']:
+                PC = getattr(self, pc)
+                self.attrs['weightedpairs'][pc] = PC.attrs['weightedpairs']
+                setattr(self,pc,PC.pairs)
+       
```

Comment:
I do not know if you want to keep the possibility of passing R1R2 (SurveyDataPairCount or SimulatedBoxPairCount instance) as additional argument of BasePairCount2PCF. If so, I would advocate keeping self.D1D2, self.D1R2, etc. as PairCountBase instance all the way (instead of getting the BinnedStatistic object pairs), and handle them in the __getstate__ and __setstate__ (using __getstate__ and ___setstate__ of PairCountBase). So that the user can access again to the PairCountBase instances R1R2, D1R2, etc. when loading BasePairCount2PCF, to do what he wants with pair counts (calculate another estimator, etc.).